agent/exec/container/adapter.go

```diff
@@ -61,15 +62,28 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 
 	dec := json.NewDecoder(rc)
 	m := map[string]interface{}{}
+	spamLimiter := rate.NewLimiter(rate.Every(1000*time.Millisecond), 1)
+
 	for {
 		if err := dec.Decode(&m); err != nil {
 			if err == io.EOF {
 				break
 			}
 			return err
 		}
-		// TODO(stevvooe): Report this status somewhere.
-		logrus.Debugln("pull progress", m)
+		if spamLimiter.Allow() {
+			if progress, ok := m["progressDetail"].(map[string]interface{}); ok {
+				// only post pull progress to log every couple of seconds
+				log.G(ctx).WithFields(logrus.Fields{
+					"id":      m["id"],
+					"status":  m["status"],
+					"current": progress["current"],
+					"total":   progress["total"],
+				}).Debug("pull in progress")
```

Comment:
Here is how it looks with `UseNumber()`. Also managed to accidentally catch a `Verifying Checksum`, so you can see how nil fields look.

```
DEBU[0015] task status updated                           method=(*Dispatcher).processUpdates module=dispatcher state.transition=ASSIGNED->PREPARING task.id=de1rx7cfow1ip1xwwd26xu1u0
DEBU[0016] pull in progress
DEBU[0017] pull in progress                              current=2702466 id=7c91a140e7a1 status=Downloading total=72873635
DEBU[0018] pull in progress                              current=21625986 id=7c91a140e7a1 status=Downloading total=72873635
DEBU[0022] pull in progress                              current=30817410 id=7c91a140e7a1 status=Downloading total=72873635
DEBU[0023] pull in progress                              current=54066306 id=7c91a140e7a1 status=Downloading total=72873635
DEBU[0024] pull in progress                              current=<nil> id=7c91a140e7a1 status=Verifying Checksum total=<nil>
DEBU[0025] pull in progress                              current=16154624 id=7c91a140e7a1 status=Extracting total=72873635
DEBU[0026] pull in progress                              current=34537472 id=7c91a140e7a1 status=Extracting total=72873635
DEBU[0027] pull in progress                              current=50692096 id=7c91a140e7a1 status=Extracting total=72873635
DEBU[0028] pull in progress                              current=65175552 id=7c91a140e7a1 status=Extracting total=72873635
DEBU[0029] pull in progress                              current=<nil> id=7c91a140e7a1 status=Digest: sha256:a97914edb6ba15deb5c5acf87bd6bd5b6b0408c96f48a5cbd450b5b04509bb7d total=<nil>
```